dotnet/targets/Xamarin.Shared.Sdk.targets

```diff
@@ -560,30 +561,39 @@
 		<WriteLinesToFile SessionId="$(BuildSessionId)" File="$(_CustomLinkerOptionsFile)" Lines="$(_CustomLinkerOptions)" Overwrite="true" />
 	</Target>
 
-	<!-- Look in the NativeReference items for frameworks that need to be added to the app bundle, and add all those frameworks to ResolvedFileToPublish (as separate files) -->
+	<!-- Look in the _FrameworkNativeReference items for frameworks that need to be added to the app bundle, and add all those frameworks to _FrameworkToPublish -->
 	<Target Name="_ComputeFrameworkFilesToPublish" DependsOnTargets="_ExpandNativeReferences;_ComputeVariables;_LoadLinkerOutput">
 		<ItemGroup>
-			<!-- Expand each framework (which are directories) into all the files in the framework -->
-			<!-- Support a 'CopyToAppBundle' metadata that can be set to 'false' to avoid copying a framework to the app bundle -->
-			<_FrameworkFilesToPublish Include="%(_FrameworkNativeReference.RootDir)%(_FrameworkNativeReference.Directory)/**/*" Condition="'%(_FrameworkNativeReference.Kind)' == 'Framework' And '%(_FrameworkNativeReference.CopyToAppBundle)' != 'false'">
-				<_FrameworkIdentity>%(RootDir)%(Directory)</_FrameworkIdentity>
-				<_FrameworkPath>$(_AppBundleFrameworksDir)\%(Filename).framework</_FrameworkPath>
-				<CopyToPublishDirectory>PreserveNewest</CopyToPublishDirectory>
-			</_FrameworkFilesToPublish>
-
-			<!-- Compute the relative path of each file in the framework relative to the framework directory -->
-			<_FrameworkFilesToPublish Update="@(_FrameworkFilesToPublish)">
-				<_FrameworkRelativePath>$([System.String]::Copy('%(Identity)').Substring($([System.String]::Copy('%(_FrameworkIdentity)').Length)))</_FrameworkRelativePath>
-			</_FrameworkFilesToPublish>
-
-			<!-- Add all the framework files to ResolvedFileToPublish -->
-			<ResolvedFileToPublish Include="@(_FrameworkFilesToPublish)">
-				<RelativePath>%(_FrameworkFilesToPublish._FrameworkPath)\%(_FrameworkFilesToPublish._FrameworkRelativePath)</RelativePath>
-				<CopyToPublishDirectory>PreserveNewest</CopyToPublishDirectory>
-			</ResolvedFileToPublish>
+			<!-- Collect the list of frameworks to publish from _FrameworkNativeReference. The ExtractBindingLibrariesStep in the linker might also add frameworks to _FrameworkToPublish -->
+			<_FrameworkToPublish Include="@(_FrameworkNativeReference)" Condition="'%(_FrameworkNativeReference.Kind)' == 'Framework' And '%(_FrameworkNativeReference.CopyToAppBundle)' != 'false'" />
+
+			<!-- Set TargetDirectory and SourceDirectory for all frameworks we have to publish -->
+			<_FrameworkToPublish Update="@(_FrameworkToPublish)">
+				<TargetDirectory>$(PublishDir)$(_AppBundleFrameworksDir)\%(Filename).framework</TargetDirectory>
```

Comment:
Is `PublishDir` a relative path? If not we may end up using Windows paths on the Mac.

Reply:
It's relative by default, but customers might set it to a full path (we don't change it).

I'll change it to make sure we're using a relative path in either case.

Reply:
@emaf I've fixed it now to use a relative path